plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/registry/SQLNativeExecutorRegistry.java

```diff
@@ -76,10 +77,15 @@ public List<SQLNativeExecutorDescriptor> getExecutors() {
      */
     @Nullable
     public SQLNativeExecutorDescriptor getExecutorDescriptor(@NotNull DBPDataSource dataSource) {
-        return executors.stream()
-            .filter(it -> it.isSupported(dataSource))
-            .findFirst()
-            .orElse(null);
+        DBPDataSourceProviderDescriptor provider = dataSource.getContainer().getDriver().getProviderDescriptor();
+        for (DBPDataSourceProviderDescriptor pd = provider; pd != null; pd = pd.getParentProvider()) {
+            for (SQLNativeExecutorDescriptor executor : executors) {
+                if (executor.getDataSourceID().equals(pd.getPluginId())) {
```

Comment:
`getPluginId`? Maybe you meant `getId`? How in the world are you submitting changes without testing them first???